rust/core-lib/src/event_context.rs

```diff
@@ -494,6 +494,32 @@ impl<'a> EventContext<'a> {
         self.editor.borrow_mut().dec_revs_in_flight();
     }
 
+    /// Returns the text to be saved, appending a newline if necessary.
+    pub(crate) fn text_for_save(&mut self) -> Rope {
+        let editor = self.editor.borrow();
+        let mut rope = editor.get_buffer().clone();
+        let rope_len = rope.len();
+
+        if rope_len < 1 {
```

Comment:
we can also return early if `self.config.save_with_newline == false`?

Reply:
Ah, true

Reply:
Pushed new version.